src/runtime_src/tools/xclbin/SectionAIEMetadata.cxx

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "SectionAIEMetadata.h"
+
+#include <boost/property_tree/json_parser.hpp>
+
+#include "XclBinUtilities.h"
+namespace XUtil = XclBinUtilities;
+
+// Static Variables / Classes
+SectionAIEMetadata::_init SectionAIEMetadata::_initializer;
+
+void 
+SectionAIEMetadata::marshalToJSON( char* _pDataSection, 
+                                   unsigned int _sectionSize, 
+                                   boost::property_tree::ptree& _ptree) const
+{
+    XUtil::TRACE("");
+    XUtil::TRACE("Extracting: AIE_METADATA");
+
+    std::unique_ptr<unsigned char> memBuffer(new unsigned char[_sectionSize + 1]);
```

Comment:
mismatched free, either std::vector or std::unique_ptr<unsigned char[]>

Reply:
Thanks for the catch.
Issue resolved and I am working on addressing this issue elsewhere in the code base.
Note: These other changes will not be part of this pull request.